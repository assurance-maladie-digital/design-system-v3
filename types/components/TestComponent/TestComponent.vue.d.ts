declare let __VLS_typeProps: {
    prop1?: string;
    number1: number;
    number2: number;
};
type __VLS_PublicProps = {
    modelValue?: any;
} & typeof __VLS_typeProps;
declare const _default: import("vue").DefineComponent<__VLS_WithDefaults<__VLS_TypePropsToOption<__VLS_PublicProps>, {
    prop1: string;
}>, {}, unknown, {}, {}, import("vue").ComponentOptionsMixin, import("vue").ComponentOptionsMixin, {
    "update:modelValue": (modelValue: any) => void;
    click: (...args: any[]) => void;
}, string, import("vue").PublicProps, Readonly<import("vue").ExtractPropTypes<__VLS_WithDefaults<__VLS_TypePropsToOption<__VLS_PublicProps>, {
    prop1: string;
}>>> & {
    onClick?: ((...args: any[]) => any) | undefined;
    "onUpdate:modelValue"?: ((modelValue: any) => any) | undefined;
}, {
    prop1: string;
}, {}>;
export default _default;

type __VLS_WithDefaults<P, D> = {
    [K in keyof Pick<P, keyof P>]: K extends keyof D ? __VLS_Prettify<P[K] & {
        default: D[K];
    }> : P[K];
};
type __VLS_Prettify<T> = {
    [K in keyof T]: T[K];
} & {};
type __VLS_NonUndefinedable<T> = T extends undefined ? never : T;
type __VLS_TypePropsToOption<T> = {
    [K in keyof T]-?: {} extends Pick<T, K> ? {
        type: import('vue').PropType<__VLS_NonUndefinedable<T[K]>>;
    } : {
        type: import('vue').PropType<T[K]>;
        required: true;
    };
};
