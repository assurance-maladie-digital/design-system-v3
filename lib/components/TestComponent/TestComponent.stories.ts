import type { Meta, StoryObj } from '@storybook/vue3'

import TestComponent from './TestComponent.vue'
import { fn } from '@storybook/test'

const meta = {
	title: 'Components/TestComponent',
	component: TestComponent,
	tags: ['autodocs'],
	args: { onClick: fn() },
	parameters: {
		layout: 'fullscreen',
	},
} satisfies Meta<typeof TestComponent>

export default meta

type Story = StoryObj<typeof meta>

export const Default: Story = {
	args: {
		number1: 1,
		number2: 2,
	},
}

export const WithDifferentNumbers: Story = {
	args: {
		number1: 3,
		number2: 4,
	},
}
