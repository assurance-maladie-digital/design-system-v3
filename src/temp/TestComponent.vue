<script setup lang="ts">
	import { computed } from 'vue'
	import { VBtn } from 'vuetify/components/VBtn'

	const props = withDefaults(defineProps<{
		prop1?: string
		number1: number
		number2: number
	}>(), {

		prop1: 'default value',
	})

	defineEmits([
		'click',
	])

	const value = defineModel< string >()
	const result = computed(() => {
		return props.number1 + props.number2
	})
</script>

<template>
	<div>
		{{ prop1 }}
	</div>
	<v-btn @click="$emit('click', result)">
		click me
	</v-btn>
	<div>
		Result : {{ result }}
	</div>
	<div>{{ value }}</div>
</template>

<style lang="scss" scoped></style>
