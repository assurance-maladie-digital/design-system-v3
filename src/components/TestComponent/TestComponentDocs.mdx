import { Meta, ArgTypes } from '@storybook/blocks';
import * as TestComponentStories from './TestComponent.stories';

<Meta of={TestComponentStories} name="TestComponent" />

# Ceci est la page de documentation du composant `TestComponent`

Ce composant est présent à des fin de test et ne doit pas être utilisé en production.
<br/>

## API
<ArgTypes of={TestComponentStories} />
