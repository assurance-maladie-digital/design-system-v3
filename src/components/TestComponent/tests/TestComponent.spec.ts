import { vuetify } from '@tests/unit/setup'
import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import TestComponent from '../TestComponent.vue'

describe('TestComponent', () => {
	it('render correctly', () => {
		const wrapper = mount(TestComponent, {
			props: {
				number1: 5,
				number2: 2,
			},
			global: {
				plugins: [vuetify],
			},
		})

		expect(wrapper.html()).toMatchSnapshot()
		expect(wrapper.text()).toContain('7')
	})
})
