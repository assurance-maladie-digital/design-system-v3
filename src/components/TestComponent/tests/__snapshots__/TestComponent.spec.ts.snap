// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TestComponent > render correctly 1`] = `
"<div>default value</div>
<button type="button" class="v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-elevated"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
  <!----><span class="v-btn__content" data-no-activator=""> click me </span>
  <!---->
  <!---->
</button>
<div> Result : 7</div>
<div></div>"
`;
